### ISSUE TITLE
When GetGrav.com is down, CMS becomes unusable

### ISSUE THREAD

**Issue by blijerd** [getgrav/grav#3383]
From time to time the website GetGrav.com is down. When this happens my complete CMS becomes unusable. It causes extreme wait times (it seems my local cms tries different JSON calls to getgrav.com).

It would be much better if my local installation doesn't rely on the uptime of GetGrav.com

**Comment by onetrev** [comment]
Agreed. Although in my case I would say there are no issues with the front-end of the site. Thankfully. But yes when getgrav.org is down (like it is right now), the Admin will not load (I get a 504 error once it finally times out).

**Comment by mahagr** [comment]
CC @rhukster @w00fz

**Comment by mahagr** [comment]
I could reproduce the issue. If I detatch ethernet cable, remove GPM cache files and visit on admin, it basically stops working.

**Comment by mahagr** [comment]
Partially fixed. The Admin still freezes if you visit Plugins or Themes pages, but otherwise it should be fine.

**Comment by pboguslawski** [comment]
> Partially fixed.

Grav should have parameter to disable any connections to public net (security reasons, compatibility with internal networks without access to public, etc.). Workaround that seems to work for Plugins and Themes is to set proxy_url to invalid localhost port i.e.:

```
gpm:
  proxy_url: 127.0.0.1:999
```

**Comment by Anke** [comment]
Hi,
excuse my question at this point. I'm looking for a flat file CMS for pure local use, e.g on USB sticks. Reading this post, I get the impression that Grav cannot be used without an internet connection. Is that so?
Anke
